test/e2e/utils.go

```diff
@@ -215,10 +215,16 @@ func deleteNodeNeworkStates() {
 }
 
 func deletePolicy(name string) {
+	By(fmt.Sprintf("deleting policy %s", name))
 	policy := &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
 	policy.Name = name
 	err := framework.Global.Client.Delete(context.TODO(), policy)
 	Expect(err).ToNot(HaveOccurred())
+	policy = nil
+	Eventually(func() *nmstatev1alpha1.NodeNetworkConfigurationPolicy {
+		framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, policy)
```

Comment:
You should check the return code and see if it is NotFound. The state of `policy` is undefined in case the function fails. Then you can drop the `policy = nil`

Reply:
Originally I was checking it the proper way with apierrors.IsNotFound, but looks like the client form operator-sdk e2e test framework is not correctly propagating the error so it does not match, we can try to debug it though, the only way I found for it to work is checking the policy, but I am not happy with it either.

Reply:
Could you try to use basic client-go instead to see if it is really an issue of the operator-sdk client? I find this disturbing

Reply:
Ok checking it correctly again works, something was not good at previous code.